### PR TITLE
added: modal component

### PR DESCRIPTION
**Pull request by rkapoor10**
Added 3 types of modal components

- Alert Modal 

alert

- Simple Modal

simple

- Confirmation Modal 

confirmation

### CODE REVIEW

File: components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    background-color: var(--white-color);
+    color: var(--primary-color);
+  }
+  
+  /* || SIMPLE DIALOG **************/
+  
+  .simple-modal {
+    display: none;
+    flex-direction: column;
+    gap: 30px;
```
Comment (dev-enforced) [inline components/modal/modal.css]: Can use rem instead of px.

File: components/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/* .modals{
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+} */
```
Comment (tushartiwari7) [inline components/modal/modal.css]: remove comments

File: components/modal/modal.css [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  .modal-list {
+    display: flex;
+    flex-direction: column;
+  }
+  
+  .modal-list-items {
+    font-size: 1.2rem;
+    display: flex;
+    align-items: center;
+    gap: 20px;
```
Comment (dev-enforced) [inline components/modal/modal.css]: Here also same as above.

**Review by dev-enforced — CHANGES_REQUESTED** [review]
The demonstration and documentation look good. Suggested some minor corrections and then you are ready to merge

File: components/modal/modal.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+//     isShowing &&
+//     "display-btn" !== e.target.className &&
+//     "dialogs" !== e.target.className
+//   ) {
+//     confirmationModal.style.display = "none";
+//     simpleModal.style.display = "none";
+//     alertModal.style.display = "none";
+//   }
+// });
+
```
Comment (tushartiwari7) [inline components/modal/modal.js:48]: remove comments please

**Review by tushartiwari7 — APPROVED** [review]
Remove Comments and rest all seems good to merge!

File: components/modal/modal.css [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    padding-bottom: 1rem;
+    border-bottom: 1px solid lightgray;
+  }
+  
+  .confirmation-list {
+    height: 10rem;
+    overflow: auto;
+  }
+  
+  input[type="radio"] {
```
Comment (paarthbajaj) [inline components/modal/modal.css:90]: Good that you have implemented an attribute selector.

**Review by paarthbajaj — APPROVED** [review]
Looks good. You can merge.

File: components/modal/modal.css (hunk 1 above)
Comment (rkapoor10) [reply components/modal/modal.css]: done

File: components/modal/modal.css (hunk 2 above)
Comment (rkapoor10) [reply components/modal/modal.css]: done 🤣

File: components/modal/modal.js (hunk 4 above)
Comment (rkapoor10) [reply components/modal/modal.js:48]: done

File: components/modal/modal.css (hunk 5 above)
Comment (rkapoor10) [reply components/modal/modal.css:90]: thanks

**Pull request merged**